addons/sourcemod/scripting/vsh/bosses/boss_whiteranger.sp

```diff
@@ -0,0 +1,380 @@
+	"vo/medic_go05.mp3",
+	"vo/medic_cheers05.mp3"
+};
+
+static char g_strUberRangerLastMan[][] = {
+	"vo/taunts/medic_taunt_kill_08.mp3"
+};
+
+static char g_strUberRangerBackStabbed[][] = {
+	"vo/medic_autodejectedtie01.mp3",
+	"vo/medic_sf12_badmagic10.mp3", 
+	"vo/medic_taunts11.mp3"
+};
+
+methodmap CUberRanger < SaxtonHaleBase
+{
+	public CUberRanger(CUberRanger boss)
+	{
+		CBraveJump abilityJump = boss.CallFunction("CreateAbility", "CBraveJump");
+		abilityJump.iJumpChargeBuild = 1;
+		
+		CRageAddCond rageCond = boss.CallFunction("CreateAbility", "CRageAddCond");
+		rageCond.flRageCondDuration = 6.0;
+		rageCond.flRageCondSuperRageMultiplier = 1.333;	//This is 7.998 seconds, close enough
+		rageCond.AddCond(TFCond_UberchargedCanteen);
+		
+		boss.iBaseHealth = 650;
+		boss.iHealthPerPlayer = 550;
+		boss.nClass = TFClass_Medic;
+		boss.iMaxRageDamage = 2500;
+		boss.bCanBeHealed = true;
+	}
+	
+	public void GetBossName(char[] sName, int length)
+	{
+		strcopy(sName, length, "White Über Ranger");
+	}
+	
+	public void GetBossInfo(char[] sInfo, int length)
+	{
+		StrCat(sInfo, length, "\nHealth: Low");
+		StrCat(sInfo, length, "\n ");
+		StrCat(sInfo, length, "\nAbilities");
+		StrCat(sInfo, length, "\n- Brave Jump (slower charge rate)");
+		StrCat(sInfo, length, "\n- Equipped with a Medi Gun");
+		StrCat(sInfo, length, "\n ");
+		StrCat(sInfo, length, "\nRage");
+		StrCat(sInfo, length, "\n- Übercharge for 6 seconds");
+		StrCat(sInfo, length, "\n- Summons a fellow Über Ranger");
+		StrCat(sInfo, length, "\n- Über Rangers are allowed to heal and über each other");
+		StrCat(sInfo, length, "\n- 200%% Rage: extends über duration to 8 seconds and summons 3 Über Rangers");
+	}
+	
+	public bool IsBossHidden()
+	{
+		return true;
+	}
+	
+	public void OnSpawn()
+	{
+		//Bosses and minions can't be overhealed, so a -max overheal attribute for the Medigun isn't needed
+		this.CallFunction("CreateWeapon", 211, "tf_weapon_medigun", 100, TFQual_Collectors, "");
+		
+		char sAttribs[128];
+		Format(sAttribs, sizeof(sAttribs), "2 ; 2.80 ; 252 ; 0.5 ; 259 ; 1.0"); 
+		int iWeapon = this.CallFunction("CreateWeapon", 173, "tf_weapon_bonesaw", 100, TFQual_Collectors, sAttribs);
+		if (iWeapon > MaxClients)
+			SetEntPropEnt(this.iClient, Prop_Send, "m_hActiveWeapon", iWeapon);
+			
+		/*
+		Vitasaw attributes:
+		
+		2: damage bonus
+		252: reduction in push force taken from damage
+		259: Deals 3x falling damage to the player you land on
+		*/
+		
+		int iColor[4];
+		
+		//For reference: 230 230 230 is the color code for TF2's white paint
+		for (int i = 0; i < 3; i++)
+			iColor[i] = 230;
+			
+		iColor[3] = 255;
+		
+		SetEntityRenderColor(this.iClient, iColor[0], iColor[1], iColor[2], iColor[3]);
+		
+		int iWearable = -1;
+		
+		//Interestingly, painting the cosmetics directly doesn't seem to make them white, so we paint it through attributes in this case
+		char sWhitePaint[16];
+		Format(sWhitePaint, sizeof(sWhitePaint), "142 ; 15132390"); 
```

Comment:
If this is a better route to paint cosmetics, is it possible to do same thing to minions with different colors?
Also value not in decimal point